src/battle-dex.ts

```diff
@@ -851,15 +851,16 @@ class ModdedDex {
 
 			let data = {...Dex.moves.get(name)};
 
-			const table = window.BattleTeambuilderTable[this.modid];
-			if (id in table.overrideAcc) data.accuracy = table.overrideAcc[id];
-			if (id in table.overrideBP) data.basePower = table.overrideBP[id];
-			if (id in table.overridePP) data.pp = table.overridePP[id];
-			if (id in table.overrideMoveType) data.type = table.overrideMoveType[id];
-			if (id in table.overrideMoveCategory) data.category = table.overrideMoveCategory[id];
-			if (id in table.overrideMoveDesc) data.desc = table.overrideMoveDesc[id];
-			if (id in table.overrideMoveShortDesc) data.shortDesc = table.overrideMoveShortDesc[id];
-
+			const overrideMoveKeys = ['accuracy', 'basePower', 'category', 'desc', 'flags', 'pp', 'shortDesc', 'target', 'type'];
+			for (const key of overrideMoveKeys) {
+				for (var i = this.gen; i < 8; i++) {
+					const table = window.BattleTeambuilderTable['gen' + i];
+					if (id in table.overrideMoveData && key in table.overrideMoveData[id]) {
+						data[key] = table.overrideMoveData[key];
```

Comment:
You missed `id` here. `table.overrideMoveData[id][key]` not ``table.overrideMoveData[key]`. Of course, that's moot because I'm rewriting this anyway.